main.py

```diff
@@ -1,21 +1,38 @@
+import numpy as np
 from graphs import *
 from geneticAlgorithm import *
 from visualizeGraph import *
+import sampleParents
+import pdb
 
 colorNum = 3
 graphPath = "graph1.gx"
 population = createInitialPopulation(graphPath, colorNum)
 bestGraph = None
 generations = 0
-# Careful! This might create infinite graphs
-while True:
-	generations+=1
-	fitnessScores, bestGraph = evaluateGeneration(population)
-	if fitness(bestGraph) == 1:
-		break
-	# Temporary
-	population = createInitialPopulation(graphPath, colorNum)
-	mutateGeneration(population)
+MAX_GENS = int(1e3)
+mating_fn = matingHalfHalf
+verbose = True
+while True or generations > MAX_GENS:
+    generations += 1
+    fitnessScores, bestGraph = evaluateGeneration(population)
+
+    if verbose:
+        print("Sum of Fitness Scores: ", np.sum(fitnessScores))
+
+    if fitness(bestGraph) == 1:
+        break
+
+    parents = sampleParents.sampleBestParents(fitnessScores)
+    newPopulation = [bestGraph]
+    parents = parents[:-1] # get rid of one set of parents to make room for bestGraph
```

Comment:
You can make this better. This works but not very efficient. I think it would be better just to replace one element at the end of the new population.

Reply:
Yeah I can do that